### PR TITLE
[BUGFIX] Batch Expectations correctly handle `date` min and max values

### PR DESCRIPTION
**Pull request by joshua-stauffer**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/integration/data_sources_and_expectations/test_canonical_expectations.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
+from datetime import datetime, timezone
+
 import pandas as pd
-import sqlalchemy.dialects.postgresql as POSTGRESQL_TYPES
```
Comment (joshua-stauffer) [inline tests/integration/data_sources_and_expectations/test_canonical_expectations.py:2]: we shouldn't import directly from sqlalchemy

Comment (tyler-hoffman) [reply tests/integration/data_sources_and_expectations/test_canonical_expectations.py:2]: Note to non-Josh folks: the postgresql types were just imported from the sqla types, so they are equivalent. Which explains mypy being fine with this change.

**Review by tyler-hoffman — APPROVED** [review]

**Comment by tyler-hoffman** [comment]
Why'd this break now (or I'm assuming as of v1?)

**Comment by joshua-stauffer** [comment]
it seems like this broke because V1 now deserializes the `min_value` and `max_value` fields on `BatchExpectations` as `datetime.date` rather than `datetime.datetime`, if they are a date rather than a timestamp.

**Pull request merged**